samza-api/src/main/java/org/apache/samza/table/remote/TableWriteFunction.java

```diff
@@ -102,43 +95,37 @@ default void putAll(List<Entry<K, V>> records) {
    * @param key key to the table record to be deleted
    */
   default void delete(K key) {
-    try {
-      deleteAsync(key).get();
-    } catch (InterruptedException | ExecutionException e) {
-      throw new SamzaException("DELETE failed for " + key, e);
-    }
+    deleteAsync(key).toCompletableFuture().join();
   }
 
   /**
    * Asynchronously delete the {@code record} with specified {@code key} from the remote store
    * @param key key to the table record to be deleted
-   * @return CompletableFuture for the delete request
+   * @return CompletionStage for the delete request
    */
-  CompletableFuture<Void> deleteAsync(K key);
+  CompletionStage<Void> deleteAsync(K key);
 
   /**
    * Delete all {@code records} with the specified {@code keys} from the remote store
    * The default implementation calls deleteAllAsync and blocks on the completion afterwards.
    * @param keys keys for the table records to be written
    */
   default void deleteAll(Collection<K> keys) {
-    try {
-      deleteAllAsync(keys).get();
-    } catch (InterruptedException | ExecutionException e) {
-      throw new SamzaException("DELETE failed for " + keys, e);
-    }
+    deleteAllAsync(keys).toCompletableFuture().join();
```

Comment:
remote the exception may break some apps which is relying on the exception type.

Reply:
+1
Should catch and re-throw